subprojects/docs-asciidoctor-extensions-base/src/test/groovy/org/gradle/docs/asciidoctor/SampleIncludeProcessorTest.groovy

```diff
@@ -223,4 +223,54 @@ include::sample[dir="src/samples",files="foo.xml[tag=bar]"]
         !content.contains('hello>')
         content.contains("&lt;child&gt;&lt;/child&gt;")
     }
+
+    def "trims indentation in samples"() {
+        given:
+        tmpDir.newFile("src/samples/build.gradle") << """
+                // Comment
+                doLast {
+                    println "hello world"
+                }
+"""
+
+        String asciidocContent = """
+= Doctitle
+:samples-dir: ${tmpDir.root.canonicalPath}
+
+include::sample[dir="src/samples",files="build.gradle[]"]
+"""
+
+        when:
+        String content = asciidoctor.convert(asciidocContent, [:])
+
+        then:
+        content.contains('// Comment\ndoLast {\n    println "hello world"\n}')
+    }
+
+    def "trims indentation in samples with tags"() {
+        given:
+        tmpDir.newFile("src/samples/build.gradle") << """
+    // tag::foo[]
+                // Comment
+                doLast {
+    // end::foo[]
+                    println "hello world"
+    // tag::foo[]
+                }
+    // end::foo[]
+"""
+
+        String asciidocContent = """
```

Comment:
I don't like tests formatted like this. Could you maybe use the stripMargin style instead?
```
                         ```ABC
                            |123
                            |456```.stripMargin()
```

Reply:
I just found that combining `.trim().stripMargin()` gives the best of both worlds:
- `trim()` allows to keep `"""` quotes on separate lines not making the first and the last line of multi-line string stand out
- `stripMargin()` allows to still use `|` margins to make it clear how much whitespace we want for each line

The result is the test code is much easier to follow visually once you recognized `trim().stripMargin()` pattern:
```
        def expectedContent = '''
            |// Comment
            |doLast {
            |    println "hello world"
            |}
        '''.trim().stripMargin()
```

Reply:
Yeah, waay better.